### DISCUSSION TITLE
Help with Docker install: Clock/Timers not working

### DISCUSSION THREAD

**Discussion by lukestein** [cpvalente/ontime#939] [Q&A] [answered]
Hi team. I'm sorry for being very incompetent with Docker, but I have successfully set up a number of other simple docker projects on my Synology NAS and for some reason I'm struggling with ontime. Any ideas appreciated.

The key problem I'm facing is that after deploying the container, while I can access and interact with Ontime via local URL, its clock always shows "00:00:00" and timers won't run, as if no time data is making it's way into the container.

![image](https://github.com/cpvalente/ontime/assets/44452336/b6bd4359-f11b-4b6e-8b44-373c204b3dd9)

![image](https://github.com/cpvalente/ontime/assets/44452336/0a486ace-c780-4809-8452-d69465bfb68a)

I installed using Synology DSM Container Manager, which has worked fine for me with other containers. I have tried to mess with no settings other than adding the volume mount, adding the TZ environmental variable, and changing the outward-facing ports from 8888/9999.

![Screenshot 2024-05-08 at 2 36 18 PM](https://github.com/cpvalente/ontime/assets/44452336/7db03a3e-2a73-4e80-9d61-3a1910df4b32)

![Screenshot 2024-05-08 at 2 36 32 PM](https://github.com/cpvalente/ontime/assets/44452336/6c3c11df-59a7-4b74-82fb-81d50db30fb3)

![Screenshot 2024-05-08 at 2 36 45 PM](https://github.com/cpvalente/ontime/assets/44452336/613231da-bb14-414f-9cf2-0242cbd07f86)

Any advice would be very much appreciated!!

**Comment by alex-Arc** [comment]
Hey @lukestein could you pull up the developer console in you browser (in chrome by pressing F12)
if you go to the network tab, filter by WS and reload then reload, you should see clock data coming in

 
![Screenshot 2024-05-08 220552](https://github.com/cpvalente/ontime/assets/19875125/31175e44-87c0-43f3-ac30-1ff751acf3cf)

**Reply by lukestein** [reply, marked as answer]
Thank you for this suggestion! The console showed that the websockets were all failing.

![image](https://github.com/cpvalente/ontime/assets/44452336/4119f16e-6b0a-4986-a8be-8817fc3ec307)

It turns out that the Synology reverse proxy apparently blocks websockets by default. My friend K. pointed me towards [this link](https://www.reddit.com/r/synology/comments/z88csq/psa_web_sockets_with_synology_is_easy/) which suggested a fix that fixed this problem in about 5 seconds. Thank you!!!

![image](https://github.com/cpvalente/ontime/assets/44452336/60fb7582-220a-423b-bae6-a05834430cb3)

![image](https://github.com/cpvalente/ontime/assets/44452336/2b978e2f-8517-4d28-9465-6ad2c63b56c2)

**Comment by cpvalente** [comment]
Hi @lukestein,

Thank you for reaching out and for the very detailed issue.

The runtime data (eg. timers) are distributed over websockets to the interface. The issue you describe is consistent with a failure of the server to deliver the websocket data to the UI.

We know that in some setups like NGINX you will need to configure a passthrough to websockets
From a quick search it seems that you may have to configure this in synology as well?
([first hit on google](https://mlohr.com/websockets-for-synology-dsm/), not sure if up to date)

This issue comes up often enough that we need to add some documentation to clarify this behaviour

Hope this heps, otherwise please follow up and we can try and help you find a solution

EDIT: just realised that, by the time I have posted this, you had figure it out. Sorry about the late reply

**Reply by lukestein** [reply]
The idea that you have anything to apologize for here is crazy… you're providing free software and timely help. Thank YOU!

If this does come up regularly, may be worth adding something to the docs (not necessarily about Synology, but maybe some troubleshooting advice on the Docker page about checking websockets) just so that anyone stuck like I was and inexpert like I am has an idea what to google.
